include/networkit/graph/Graph.hpp

```diff
@@ -1363,7 +1363,155 @@ class Graph final {
+                        outEdgeWeights[u].insert(outEdgeWeights[u].end(),
+                                                 G.outEdgeWeights[u].begin(),
+                                                 G.outEdgeWeights[u].end());
+                    } else {
+                        // we are undirected, so no need to write anything into inEdgeWeights
+                        outEdgeWeights[u] =
+                            std::vector<edgeweight>(outEdges[u].size(), defaultEdgeWeight);
+                    }
+                }
+                if (G.hasEdgeIds() && edgesIndexed) {
+                    // copy both out and in edges ids into our new outEdgesIds
+                    outEdgeIds[u].reserve(G.outEdgeIds[u].size() + G.inEdgeIds[u].size());
+                    outEdgeIds[u].insert(outEdgeIds[u].end(), G.outEdgeIds[u].begin(),
+                                         G.outEdgeIds[u].end());
+                }
+            }
+            for (node u = 0; u < z; ++u) {
+                // this is necessary to avoid multi edges, because both u -> v and v -> u can exist
+                // in G
+                count edgeSurplus = 0;
+                for (count i = 0; i < G.inEdges[u].size(); ++i) {
+                    node v = G.inEdges[u][i];
+                    bool alreadyPresent = false;
+                    for (count j = 0; j < G.outEdges[u].size(); ++j) {
+                        if (v != G.outEdges[u][j])
+                            continue; // the edge already exists as an out edge
+                        alreadyPresent = true;
+                        if (u != v) {
+                            ++edgeSurplus;
+                            if (weighted) // we need combine those edges weights when making it a
+                                          // single edge
+                                outEdgeWeights[u][j] =
+                                    G.isWeighted()
+                                        ? edgeMerger(G.inEdgeWeights[u][i], G.outEdgeWeights[u][j])
+                                        : edgeMerger(defaultEdgeWeight, defaultEdgeWeight);
+                            if (G.hasEdgeIds() && edgesIndexed)
+                                outEdgeIds[u][j] = std::min(G.inEdgeIds[u][i], G.outEdgeIds[u][j]);
+                        }
+                        break;
+                    }
+                    if (!alreadyPresent) { // an equivalent out edge wasn't present so we add it
+                        outEdges[u].push_back(v);
+                        if (weighted)
+                            outEdgeWeights[u].push_back(G.isWeighted() ? G.inEdgeWeights[u][i]
+                                                                       : defaultEdgeWeight);
+                        if (G.hasEdgeIds() && edgesIndexed)
+                            outEdgeIds[u].push_back(G.inEdgeIds[u][i]);
+                    }
+                }
+                m -= edgeSurplus;
```

Comment:
If we use a parallel loop, this can be updated using an atomic operation or an omp reduction.